### DISCUSSION TITLE
How can I cast a variable I'm sending to an RPC (function)

### DISCUSSION THREAD

**Discussion by nruffilo** [supabase/supabase#1183] [Questions] [answered]
I have a function with the signature: CREATE OR REPLACE FUNCTION CompleteQuest(questId bigint)
The problem is, when I run it using .rpc("CompleteQuest", {questId: completedQuestId}) I get "No function matches the given name and argument types. You might need to add explicit type casts.","details":null,"code":"42883","message":"function public.CompleteQuest(questId => text) does not exist"

the completedQuestId is an INT.  I've tried creating a new variable and using parseInt() but it always seems to send as text.  How can I mark what the variable type is that I'm passing?  Or, in the function, do I need to assume only text will be passed and cast it in the PL/SQL?

Thank you!

Additional info: 
![image](https://user-images.githubusercontent.com/1260513/115130815-c0ac8100-9fc0-11eb-87f9-1e6b9b8334c0.png)
The param is being sent as an INT, but somewhere the RPC is casting it as TEXT.

**Comment by steve-chavez** [comment, marked as answer]
You need to double quote the function name if you use a camel case identifier, like:

```sql
CREATE OR REPLACE FUNCTION "CompleteQuest"("questId" bigint)
```

Otherwise postgres will convert the identifier to lowercase. 

(Also quote the parameter names if they have uppercase chars)

**Reply by nruffilo** [reply]
Thank you - that did the trick.  I appreciate your quick and complete response.
